src/background/index.ts

```diff
@@ -0,0 +1,68 @@
+import browser, { WebRequest } from 'webextension-polyfill'
+import type { Locale, MessageType } from '../types'
+
+declare const chrome: any
+let locale: Locale = null
+
+function updateLocale(value: Locale) {
+  if (value === locale) return
+  locale = value || null
+  browser.storage.local.set({ locale })
+  browser.browserAction.setBadgeText({ text: locale })
+}
+
+async function sendMessageToActiveTabs(message: MessageType) {
+  const active = await browser.tabs.query({ active: true, currentWindow: true })
+  for (const tab of active) {
+    if (tab.id) browser.tabs.sendMessage(tab.id, message)
+  }
+}
+
+async function getCurrentLocale() {
+  const active = await browser.tabs.query({ active: true, currentWindow: true })
+  for (const tab of active) {
+    if (tab.id) browser.tabs.sendMessage(tab.id, { type: 'get' })
```

Comment:
On testing locally, there seems to be another bug:

1. Download the extension and install it
2. Set the locale on one page, say "Tab A"
3. Switch to another page that was already open when the extension was installed, say "Tab B"
4. Notice that the popup's badge and locale value do not set to null as expected. This is because the content script is not yet initialized on "Tab B" so we never get a response for this message.

Reply:
Done